snap/info_snap_yaml.go

```diff
@@ -587,6 +655,20 @@ func bindUnscopedPlugs(snap *Info, strk *scopedTracker) {
 			hook.Plugs[plugName] = plug
 			plug.Hooks[hookName] = hook
 		}
+
+		for _, component := range snap.Components {
```

Comment:
This part is a little confusing to me. From that I can tell, a plug is treated as "unscoped" if it isn't listed underneath an `app` or a `hook` in the `snap.yaml` file. These "unscoped" plugs are then attatched to all apps and hooks. I followed this pattern for component hooks as well, but I'm not sure I understand it.